src/library/scala/collection/mutable/Map.scala

```diff
@@ -71,6 +79,9 @@ trait MapOps[K, V, +CC[X, Y] <: MapOps[X, Y, CC, _], +C <: MapOps[K, V, CC, C]]
     with Growable[(K, V)]
     with Shrinkable[K] {
 
+  override def withFilter(p: ((K, V)) => Boolean): MapOps.WithFilter[K, V, Iterable, CC] = new MapOps.WithFilter(this, p)
+  override def ++:[B >: (K, V)](that: IterableOnce[B]): Iterable[B] = iterableFactory.from(that) ++ coll
```

Comment:
That recurring override is really painful… This also means that implementers of custom `Set` and `Map` will have to write this override...